TourMate/TourMate/Frontend/Plan/Views/PlanViews/PlanViewModel.swift

```diff
@@ -47,34 +47,11 @@ class PlanViewModel<T: Plan>: ObservableObject {
             .store(in: &cancellableSet)
     }
 
-    func fetchPlan() async {
-        self.isLoading = true
-        let (plan, errorMessage) = await planService.fetchPlan(withPlanId: plan.id)
-
-        guard errorMessage.isEmpty else {
-            self.isLoading = false
-            self.hasError = true
-            return
-        }
-
-        // no plans fetched
-        guard plan != nil else {
-            self.isDeleted = true
-            self.isLoading = false
-            return
-        }
+    func fetchPlanAndListen() async {
+        planService.planEventDelegate = self
 
-        // cannot cast fetched Plan into specific T-Plan
-        guard let plan = plan as? T,
-              errorMessage.isEmpty else {
-            self.isLoading = false
-            self.hasError = true
-            return
-        }
-
-        await updatePublishedProperties(plan: plan)
-
-        self.isLoading = false
+        self.isLoading = true
+        await planService.fetchPlanAndListen(withPlanId: plan.id)
```

Comment:
I understand that `planService` will call the delegate's `update()` function which have a `self.isLoading = false` but it was quite hard to trace and follow the logic. Is is possible to put the change here like this?
```suggestion
        self.isLoading = true
        await planService.fetchPlanAndListen(withPlanId: plan.id)
        self.isLoading = false
```

Reply:
Ok, I think this is fine.